include/open62541pp/types/Variant.h

```diff
@@ -72,54 +69,35 @@ class Variant : public TypeWrapper<UA_Variant, UA_TYPES_VARIANT> {
         return variant;
     }
 
-    /// Create Variant from array (no copy if assignable without conversion).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<T> array) {
+    /// Create Variant from array (no copy if assignable).
+    template <typename ArrayLike>
+    [[nodiscard]] static Variant fromArray(ArrayLike&& array) {
+        using ValueType = typename std::remove_reference_t<ArrayLike>::value_type;
+        constexpr bool isMutable = detail::IsMutableContainer<ArrayLike>::value;
+        constexpr bool isContiguous = detail::IsContiguousContainer<ArrayLike>::value;
         Variant variant;
-        if constexpr (detail::isRegisteredType<T>) {
+        if constexpr (isMutable && isContiguous && detail::isRegisteredType<ValueType>) {
             variant.setArray(array);
         } else {
             variant.setArrayCopy(array);
         }
         return variant;
     }
```

Comment:
This is not possible with the current overloads. The key are following overloads:
- `setArrayCopy(array)` will try a conversion if the data type is not known
- `setArrayCopy(array, dtype)` just copies the data because because `UA_DataType` is already known

Maybe we can refactor this to avoid the code duplication.

Reply:
Ah yes 👍